src/category_theory/limits/shapes/pullbacks.lean

```diff
@@ -37,11 +37,11 @@ The type of objects for the diagram indexing a pullback, defined as a special ca
 abbreviation walking_cospan : Type v := wide_pullback_shape walking_pair
 
 /-- The left point of the walking cospan. -/
-abbreviation walking_cospan.left : walking_cospan := some walking_pair.left
+@[pattern] abbreviation walking_cospan.left : walking_cospan := some walking_pair.left
```

Comment:
I've actually never used `@[pattern]`. Could you explain why this is necessary/desirable/useful?

Reply:
This makes it possible to use these abbreviations in pattern-matching, so we can write:
```lean
def pullback_proj : Π (j : walking_cospan), (pullback_obj f g) ⟶ (limits.cospan f g).obj j
| left  p := p.val.1
| right p := p.val.2
| one   p := f p.val.1
```

rather than:
```lean
def pullback_proj : Π (j : walking_cospan), (pullback_obj f g) ⟶ (limits.cospan f g).obj j
| (some left ) p := p.val.1
| (some right) p := p.val.2
| none         p := f p.val.1
```

Without this attribute, the first version is interpreted as trying to bind a variable named `left`, etc., so actually the first pattern in the list matches all possible values. (In the second, when we say `left` or `right` they're interpreted as `walking_pair.left` or `walking_pair.right`.)

I'm not sure the pattern-matching will survive the refactor you outlined in the Zulip thread -- I'm just about to try that refactor out. But either way it's sensible to want to be able to use these names in patterns -- the fact that they're implemented as `some walking_pair.left` etc. feels like it's an implementation detail of `walking_cospan`.